packages/app/src/cli/models/extensions/specifications/app_config_webhook.ts

```diff
@@ -17,81 +17,49 @@ const spec = createConfigExtensionSpecification({
 
 // Transform methods
 export function transformWebhookConfig(content: object) {
-  const webhooks = getPathValue(content, 'webhooks')
+  const webhooks = getPathValue(content, 'webhooks') as WebhookConfig
   if (!webhooks) return content
 
   // normalize webhook config with the top level config
   const webhookSubscriptions = []
-  const {topics, subscriptions, uri} = webhooks as WebhookConfig
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const {api_version, subscriptions = []} = webhooks
 
-  if (uri && topics?.length) {
-    for (const topic of topics) {
-      webhookSubscriptions.push({
-        topic,
-        uri,
-      })
-    }
+  // eslint-disable-next-line no-warning-comments
+  // TODO: pass along compliance_topics once we're ready to store them in its own module
+  for (const {uri, topics, compliance_topics: _, ...data} of subscriptions) {
+    webhookSubscriptions.push(topics.map((topic) => ({uri, topic, ...data})))
   }
 
-  if (subscriptions?.length) {
-    for (const {path, uri: localUri, ...subscription} of subscriptions) {
-      // we can assume this is valid from earlier validation, and local URI will overwrite top level if there is any
-      const subscriptionConfig = {
-        uri: localUri || uri,
-        ...subscription,
-      }
-
-      if (path) {
-        subscriptionConfig.uri = `${subscriptionConfig.uri}${path}`
-      }
-
-      webhookSubscriptions.push(subscriptionConfig)
-    }
-  }
-  return webhookSubscriptions.length > 0 ? {subscriptions: webhookSubscriptions} : {}
+  return webhookSubscriptions.length > 0 ? {subscriptions: webhookSubscriptions.flat(), api_version} : {api_version}
 }
 
+// link --> TOML
 export function transformToWebhookConfig(content: object) {
+  let webhooks = {}
+  const apiVersion = getPathValue(content, 'api_version') as string
+  webhooks = {...(apiVersion ? {webhooks: {api_version: apiVersion}} : {})}
   const serverWebhooks = getPathValue(content, 'subscriptions') as NormalizedWebhookSubscription[]
-  const frequencyMap: {[key: string]: number} = {}
-  serverWebhooks.forEach((item) => {
-    frequencyMap[item.uri!] = (frequencyMap[item.uri!] || 0) + 1
-  })
-  const maxCount = Math.max(...Object.values(frequencyMap))
-  const defaultUri = Object.keys(frequencyMap).find((key) => frequencyMap[key] === maxCount)
+  if (!serverWebhooks) return webhooks
 
-  const topics: string[] = []
-  const subscriptions: NormalizedWebhookSubscription[] = []
+  const webhooksSubscriptionsConfig: WebhookConfig['subscriptions'] = []
 
-  for (const item of serverWebhooks) {
-    if (item.uri === defaultUri && !item.sub_topic && !item.include_fields && !item.metafield_namespaces) {
-      topics.push(item.topic)
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  for (const {uri, topic, sub_topic, ...data} of serverWebhooks) {
+    const currSubscription = webhooksSubscriptionsConfig.find((sub) => sub.uri === uri && sub.sub_topic === sub_topic)
```

Comment:
This is basically our uniqueness check. A subscription is deemed unique if at least one of the `topic`, `uri`, or `sub_topic` is different. This is the same way the API does it, we're mimicking that logic here